[ConnectedMachine] release stable version 2022-12-27

yaotongms opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description

This release is targeting version 2022-12-27 of ConnectedMachine. You will see all commands from the previous version with the following new commands:

- Install-AzConnectedMachinePatch: install patches on an azure-enabled server
- Invoke-AzConnectedAssessMachinePatch: assess patches on an azure-enabled server
- Get-AzConnectedExtensionMetadata: display metadata of all extensions, or display metadata of extensions with specific version

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

YanaXu commented:
Hi @yaotongms ，you're using an old version autorest which will revert the correct update to wrong. Please run **autorest --reset** to clean your autorest cache and run autorest and build again.

yaotongms commented:
> Hi @yaotongms ，you're using an old version autorest which will revert the correct update to wrong. Please run **autorest --reset** to clean your autorest cache and run autorest and build again.

autorest rebuilt in the latest commit

YanaXu commented:
Hi @yaotongms , 

Please run **build-module.ps1** again and commit all updated files. 2 new files are missing.
And the noun of the new cmdlets does not start with "AzConnectedMachine". Could you please update them?

YanaXu commented on src/ConnectedMachine/docs/Update-AzConnectedMachine.md:
````diff
@@ -81,8 +84,54 @@ Update a machine to clean a field
+Accept wildcard characters: False
+```
+
+### -AgentUpgradeEnableAutomaticUpgrade
````
better to rename it to "AgentUpgradeEnableAutomatic"

yaotongms replied:
changed in the latest commit

YanaXu commented on src/ConnectedMachine/docs/Install-AzConnectedMachinePatch.md:
````diff
@@ -0,0 +1,435 @@
+ [-DefaultProfile <PSObject>] [-AsJob] [-NoWait] [-Confirm] [-WhatIf] [<CommonParameters>]
+```
+
+### Install
````
"Not-Expanded" parameter sets can be removed.

yaotongms replied:
changed in the latest commit

yaotongms commented:
> Hi @yaotongms ,
> 
> Please run **build-module.ps1** again and commit all updated files. 2 new files are missing. And the noun of the new cmdlets does not start with "AzConnectedMachine". Could you please update them?

The 2 new files related to the network profile command is not supposed to be included in this release, thus got removed.

yaotongms commented:
Changelog:

Version 0.6.0
- Updated the API version to stable 2022-12-27
- Added cmdlet Install-AzConnectedMachinePatch, Invoke-AzConnectedAssessMachinePatch and Get-AzConnectedExtensionMetadata  
- Expanded parameters for installing patches for better usage
